src/Heatmap/HeatMapInterpolationPoints.swift

```diff
@@ -0,0 +1,340 @@
+    /// - Returns: A list containing just the numerator and denominator
+    private func findIntensity(lat: Double, long: Double, n: Double) -> [Double] {
+        var numerator: Double = 0
+        var denominator: Double = 0
+        for point in self.data {
+            let dist = self.distance(
+                lat1: lat,
+                long1: long,
+                lat2: point.point().y * normalLat,
+                long2: point.point().x * normalLong
+            )
+            let distanceWeight = pow(dist, Double(n))
+            if distanceWeight == 0 {
+                continue
+            }
+            numerator += (Double(point.intensity) / distanceWeight)
+            denominator += (1 / distanceWeight)
+        }
+        return [numerator, denominator]
+    }
+
+    /// A helper function that finds the minimum and maximum longitude and latitude values that still contains a powerful enough
+    /// intensity that it should be included in the data set
+    ///
+    /// - Parameters:
+    ///     - input: A list of points that are in a cluster.
+    ///     - n: The power value that determines the drop off rate of intensities with respect to the distance from given points
+    /// - Returns: A list of four integers representing the minimum and maximum longitude and latitude values
+    private func findBounds(input: [CLLocationCoordinate2D], n: Double) -> [Int] {
+
+        // Initialize the boundary values to something that must be updated immediately
+        // 0: min lat, 1: min long, 2: max lat, 3: max long
+        var ans = [0x7fffffff, 0x7fffffff, -0x7fffffff, -0x7fffffff]
+        for coord in input {
+            ans[0] = min(ans[0], Int(coord.latitude * 10))
+            ans[1] = min(ans[1], Int(coord.longitude * 10))
+            ans[2] = max(ans[2], Int(coord.latitude * 10))
+            ans[3] = max(ans[3], Int(coord.longitude * 10))
+        }
+        return ans
+    }
+
+    /// Generates several heat maps based on the clusters with points not found in the data set interpolated by the inverse distance
+    /// means interpolation algorithm and displays the heat maps on the given map
+    ///
+    /// - Parameters:
+    ///   - mapView: The map that we want to display the heat maps on.
+    ///   - n: The n-value, determining the range of influence the intensities found in the given data set has.
+    public func generateHeatMaps(mapView: GMSMapView, n: Double) -> [GMUWeightedLatLng] {
+
+        // It doesn't make too much sense to do interpolation on an n-value of less than 2 or
+        // greater than 2.5; when n is higher, the denominator increases quicker, meaning the
+        // overall value falls quicker as the distances increase, implying that a low n value will
+        // query far too many points
+        if n < 2 || n > 2.5 {
+            return []
+        }
+        heatMapPoints.removeAll()
+        heatMapLayer.map = nil
```

Comment:
I suggest removing this instance variable altogether and defer the responsibility of adding a heatmap to the map to the client.

Reply:
Done! All heat map related variables in the interpolation class have now been removed.